placement/types.go

```diff
@@ -414,12 +414,11 @@ type Options interface {
 type ShardStateType int
```

Comment:
Perhaps we should also rename the type to something else? Feels like this is describing how the placement should be generated rather than the type of shard states. Do you think `ShardStateMode` or `PlacementMode` or `ShardStateConstraint` would be more fitting?

Reply:
I think PlacementMode sounds better, also we can do StableShardStateOnly and AllowTransitionalShardStates for the enums

Reply:
Actually I'm more leaning towards ShardStateMode now as this mainly controls shard states